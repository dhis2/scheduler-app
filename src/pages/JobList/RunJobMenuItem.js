import React from 'react'
import { func, string } from 'prop-types'
import { connect } from 'react-redux'
import { MenuItem } from '@dhis2/ui-core'
import { actions } from '../../data/modal'
import { modalTypes } from '../../components/Modal'

export const DumbRunJobMenuItem = ({ id, showModal }) => (
    <MenuItem
        dense
        onClick={() => showModal({ type: modalTypes.RUN_JOB, props: { id } })}
        label="Run manually"
    />
)

DumbRunJobMenuItem.propTypes = {
    id: string.isRequired,
    showModal: func.isRequired,
}

const mapDispatchToProps = {
    showModal: actions.showModal,
}

export default connect(
    null,
    mapDispatchToProps
)(DumbRunJobMenuItem)
