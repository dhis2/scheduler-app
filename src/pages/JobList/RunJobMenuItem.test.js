import React from 'react'
import { shallow, mount } from 'enzyme'
import RunJobMenuItem from './RunJobMenuItem'

describe('<RunJobMenuItem>', () => {
    it('renders without errors', () => {
        shallow(<RunJobMenuItem id="id" />)
    })

    it('shows the modal when MenuItem is clicked', () => {
        const wrapper = mount(<RunJobMenuItem id="id" />)

        expect(wrapper.find('RunJobModal')).toHaveLength(0)

        wrapper.find('a').simulate('click')

        expect(wrapper.find('RunJobModal')).toHaveLength(1)
    })
})
