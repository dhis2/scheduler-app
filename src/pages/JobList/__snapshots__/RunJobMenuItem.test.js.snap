// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RunJobMenuItem> renders correctly 1`] = `
<Fragment>
  <MenuItem
    dataTest="dhis2-uicore-menuitem"
    dense={true}
    label="Run manually"
    onClick={[Function]}
  />
</Fragment>
`;
